run_test.go

```diff
@@ -210,14 +210,24 @@ func TestRunsForceCancel(t *testing.T) {
 	t.Run("after a normal cancel", func(t *testing.T) {
 		// Request the normal cancel
 		err := client.Runs.Cancel(ctx, rTest.ID, RunCancelOptions{})
-		assert.NoError(t, err)
+		require.NoError(t, err)
 
-		// Allow time for the async cancel request to operate.
-		time.Sleep(5 * time.Second)
+		for i := 1; ; i++ {
+			// Refresh the view of the run
+			rTest, err = client.Runs.Read(ctx, rTest.ID)
+			require.NoError(t, err)
 
-		// Refresh the view of the run
-		rTest, err = client.Runs.Read(ctx, rTest.ID)
-		assert.NoError(t, err)
+			// Check if the timestamp is present.
+			if !rTest.ForceCancelAvailableAt.IsZero() {
+				break
+			}
+
+			if i > 5 {
```

Comment:
Since we're now in a loop I would set this to 30. We check every second so we'll not wait longer then needed, but it could prevent any flaky results when we run the test and TFE has some coincidental peak load at that moment.

Reply:
This is only to paper over the time it takes for Sidekiq to pop the job off the queue and mark the run as canceling (it populates a timestamp). It doesn't need to cover the work going all the way through to the TBW and actually doing the cancel. We could do 30s, I just figured 5 was far more than enough in any normal case and anything longer indicates a serious performance degradation in TFE. Still think we should?

Reply:
Guess it doesn't matter that much. I tent to use 30s in tests because I'm fine with some delays and maybe I'm testing this locally on a busy macbook. So if you want to keep it at 5s that fine too.

Reply:
Makes sense, no reason not to I suppose. I bumped it up.